lib/external/sentry.js

```diff
@@ -1,7 +1,93 @@
 const { inspect } = require('util');
 const { isBlank } = require('../util/util');
 
-const sensitiveEndpoints = ['/users/reset/verify', '/users/(.*?)/password'];
+
+// Endpoints where query strings are sensitive and should be removed
+const sensitiveEndpoints = [
+  ['GET', '/users[^/]'], // ?q=<part of name or email>
+  ['GET', '/projects/([^/]*)/forms/([^/]*)/submissions.csv'], // ?key=passphrase for managed encryption
+  ['GET', '/projects/([^/]*)/forms/([^/]*)/draft/submissions.csv'], // ?key=passphrase for managed encryption
+  ['GET', '/projects/([^/]*)/formList'], // ?st=token for enketo public link access (showing form)
+  ['HEAD', '/projects/([^/]*)/formList'], // ?st=token for enketo public link access (showing form)
+  ['POST', '/projects/([^/]*)/submission'], // ?st=token for enketo public link access (submitting)
+];
+
+const isSensitiveEndpoint = (request) => {
+  for (const [method, endpoint] of sensitiveEndpoints) {
+    if (request.method === method && request.url.match(endpoint)) {
+      return true;
+    }
+  }
+  return false;
+};
+
+const filterTokenFromUrl = (url) => {
+  // remove user token from any URL
+  if (url.match('/v1/key/(.*?)')) {
+    return url.replace(/v1\/key\/([^/]*)/, 'v1/key/[FILTERED]');
+  }
+
+  // remove draft token from any URL
+  if (url.match('/v1/test/(.*?)')) {
+    return url.replace(/v1\/test\/([^/]*)/, 'v1/test/[FILTERED]');
+  }
+
+  // return original URL if there are no tokens
+  return url;
+};
+
+const sanitizeEventRequest = (event) => {
+  /* eslint-disable no-param-reassign */
+  if (event.request) {
+
+    // clear out all cookie values
+    if (event.request.headers && event.request.headers.cookie) {
+      const cookies = event.request.headers.cookie.split('; ')
+        .map((cookie) => cookie.slice(0, cookie.indexOf('=')))
+        .join(', ');
+
+      event.request.headers.cookie = cookies;
+    }
+
+    // clear out cookie values not stored in request header
+    if (event.request.cookies)
+      event.request.cookies = null;
```

Comment:
for all of these null-out assignments, i wonder if it would be a better idea to do something like eg `event.request.cookies.replace(/./g, '*')` or even just `'***'` so we know there _was_ something there?

Reply:
I like the idea of clearly showing when something was filtered out by us vs. just not set for some reason.

Reply:
i'm having a hard time detecting empty objects and not overwriting them with '***' though (because of some weird nuance of javascript)